Dockerfile

```diff
@@ -33,7 +33,11 @@ WORKDIR /
 # OpenSSL is required so wget can query HTTPS endpoints for health checking.
 RUN apk add --no-cache --update ca-certificates openssl
 
-USER 1001:1001
+# Install dockerize to provide the way of config parametrization
+ENV DOCKERIZE_VERSION v0.6.1
+RUN wget https://github.com/jwilder/dockerize/releases/download/$DOCKERIZE_VERSION/dockerize-linux-amd64-$DOCKERIZE_VERSION.tar.gz \
+    && tar -C /usr/local/bin -xzvf dockerize-linux-amd64-$DOCKERIZE_VERSION.tar.gz \
```

Comment:
The kind of a problem we can't solve on our own. Looks like we need to search for a better solution.

Reply:
The [templating code](https://github.com/jwilder/dockerize/blob/master/template.go) of Dockerize is only 200 lines, since it's basically Go's `text/template` with some helper functions added.
On first sight, only other feature from Dockerize that could be helpful for Dex is waiting for other dependencies (`-wait` , `-timeout` & co). And that can usually be solved with a crash loop.

Would it be an option to glue together ourselves Go templating with helpers from a well-maintained library like https://github.com/Masterminds/sprig in Dex? 
Could work like this for the user:
`$ dex serve config.yaml`
`$ dex serve --template config.tmpl`

Sprig already includes `env` and `default` [functions](http://masterminds.github.io/sprig/), similar to Dockerize.

Reply:
FYI, if you want to use dockerize, I recommend using its fork https://github.com/powerman/dockerize
This fork was created by an active contributor who was not satisfied with Jwilder's project management, communication, and so on.
Since then, a lot of improvements were done in Powerman's fork, incl. arm builds, for example.
The original dockerize seems to be an abandoned project.

Reply:
I'm familiar with that fork, but I'm not sure I'm quite happy with the democratized maintenance model to implements:

> Everyone who has contributed to the project may become a collaborator - just ask for it in PR comments after your PR has being merged.

Also, we really don't need all that functionality, so I'm happy with a simple templating solution. My only concern with gotemplate is its size.